velox/functions/prestosql/aggregates/tests/SetAggTest.cpp

```diff
@@ -155,6 +155,90 @@ TEST_F(SetAggTest, groupBy) {
       {expected});
 }
 
+TEST_F(SetAggTest, decimal) {
+  // Test with short decimal
+  auto type = DECIMAL(6,2);
+
+  auto data = makeRowVector({
+      makeFlatVector<int64_t>({1000, 2000, 3000, 4000, 5000, 3000, 4000, 2000, 6000, 7000}, type),
```

Comment:
Please add values at the boundaries on int64_t. Also add -ve numbers in the tests as well.